docs/features/secret-store/provider/key-per-file.md

```diff
@@ -0,0 +1,88 @@
+---
+title: "KeyPerFile secret provider"
+layout: default
+---
+
+# KeyPerFile secret provider
+The _KeyPerFile_  secret provider allows working with Docker secrets.  When using Docker secrets in Docker Swarm, the secrets are injected in the Docker container as files.  
+The _KeyPerFile_ secret provider provides access to those secrets via the secret store.
+
+## Installation
+Adding secrets from the User Secrets manager into the secret store requires following package:
+
+```shell
+PM > Install-Package Arcus.Security.Providers.KeyPerFile
+```
+
+## Configuration
+After installing the package, the addtional extensions becomes available when building the secret store.
+
+```csharp
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        CreateHostBuilder(args).Build().Run();
+    }
+
+    public static IHostBuilder CreateHostBuilder(string[] args)
+    {    
+        return Host.CreateDefaultBuilder(args)
+                   .ConfigureSecretStore((context, config, builder) =>
+                    {
+                        // Adds the secrets that exist in the "/run/secrets" directory to the ISecretStore
+                        // Docker secrets are by default mounted into the /run/secrets directory
+                        // when using Linux containers on Docker Swarm.
+                        builder.AddKeyPerFile(directoryPath: "/run/secrets", optional: true);
+                    })
+                    .ConfigureWebHostDefaults(webBuilder => webBuilder.UseStartup<Startup>());
+    }
+}
+```
+
+## Retrieving secrets
+
+Suppose you have the following docker-compose file:
+
+```yaml
+version: '3.8'
```

Comment:
Any change we can set up a Docker project and make an integration test for this? (Maybe diff PR if too much work).

Reply:
Let's do seperate PR indeed so you can have a look at it and we don't put the burden on @fgheysels indeed, if ok?

Reply:
How do you see that implemented ?

Reply:
Something like starting the Docker project with ebmedded secrets beforehand, or don't we need a Docker project to test this? Can we just write directly to a file like the UserSecrets? Than it would be even more easy.